src/Features/CSharpTest/ConvertProgram/ConvertToTopLevelStatementsRefactoringTests.cs

```diff
@@ -224,23 +227,203 @@ public async Task TestConvertToTopLevelStatementWithTopLevelStatementPreferenceS
         // if the user has the analyzer suppressed, then we want to supply teh refactoring.
         await new VerifyCS.Test
         {
-            TestCode = @"
-internal class Program
-{
-    private static void $$Main(string[] args)
+            TestCode = """
+            internal class Program
+            {
+                private static void $$Main(string[] args)
+                {
+                    System.Console.WriteLine(0);
+                }
+            }
+            """,
+            FixedCode = """
+            System.Console.WriteLine(0);
+
+            """,
+            LanguageVersion = LanguageVersion.CSharp10,
+            TestState = { OutputKind = OutputKind.ConsoleApplication },
+            Options =
+            {
+                { CSharpCodeStyleOptions.PreferTopLevelStatements, true, NotificationOption2.None },
+            }
+        }.RunAsync();
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/78002")]
+    public async Task TestPreserveStatementDirectives1()
     {
-        System.Console.WriteLine(0);
+        await new VerifyCS.Test
+        {
+            TestCode = """
+                using System;
+
+                class Program
+                {
+                    static void $$Main(string[] args)
+                    {
+                #if true
+                        Console.WriteLine("true");
+                #else
+                        Console.WriteLine("false");
+                #endif
+                    }
+                }
+                """,
+            FixedCode = """
+                using System;
+
+                #if true
+                Console.WriteLine("true");
```

Comment:
it feels a bit weird for the statements to be differently tabbed

Reply:
Yes. Though it matches our philosophy of not touching pp contents as we do not actually know what they represent.